repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/RunRepositoryTestImpl.scala

```diff
@@ -0,0 +1,44 @@
+package cromwell.pipeline.datastorage.dao.repository
+import cromwell.pipeline.datastorage.dto.Run
+import cromwell.pipeline.model.wrapper.{ RunId, UserId }
+
+import scala.collection.mutable
+import scala.concurrent.Future
+
+class RunRepositoryTestImpl extends RunRepository {
+
+  private val runs: mutable.Map[RunId, Run] = mutable.Map.empty
+
+  def getRunByIdAndUser(runId: RunId, userId: UserId): Future[Option[Run]] = {
+    val run = for {
+      (id, entity) <- runs if id == runId && entity.userId == userId
+    } yield entity
+    Future.successful(run.headOption)
+  }
+
+  def deleteRunById(runId: RunId): Future[Int] = {
+    if (runs.contains(runId)) runs -= runId
```

Comment:
```suggestion
    runs -= runId
```
we don't need this check

Reply:
@myazinn done